tests/integration/release/test_release.py

```diff
@@ -112,92 +131,332 @@ def test_full_release_noising(
+                            lambda data_: pd.Series(
+                                (
+                                    get_missingness_data(data_).values
+                                    == Dataset.is_missing(get_noised_data(data_)).values
+                                ).all()
+                            )
+                        )
+                        with check:
+                            assert missingness_correct.all().compute()
+
+                        counts = data.map_partitions(
+                            lambda data_: get_column_noising_counts(
+                                data_,
+                                config,
+                                dataset_schema.name,
+                                noise_type.name,
+                                column.name,
+                            ),
+                            meta=pd.DataFrame(
+                                columns=[
+                                    "numerator",
+                                    "denominator",
+                                    "expected_numerator",
+                                    "missing_data_not_missing",
+                                ],
+                                dtype=int,
+                            ),
+                        )
+                        total_counts = counts.sum().compute()
+                        with check:
+                            assert total_counts["missing_data_not_missing"] == 0
+                        numerator = total_counts["numerator"]
+                        denominator = total_counts["denominator"]
+                        # we can get no eligible rows for unit number in acs RI data
+                        # because it's a sparse column and acs is small
+                        if column == COLUMNS.unit_number and denominator == 0:
+                            continue
+                        expected_numerator = total_counts["expected_numerator"]
+                        with check:
+                            fuzzy_checker.fuzzy_assert_proportion(
+                                name=noise_type.name,
+                                observed_numerator=numerator,
+                                observed_denominator=denominator,
+                                target_proportion=expected_numerator / denominator,
+                                name_additional=f"{dataset_name}_{column}_{noise_type}",
+                            )
+
+            if noise_type == NOISE_TYPES.duplicate_with_guardian:
+                # noising after duplicate_with_guardian should be done on prenoised data
+                # since it duplicates simulant ID which must be unique to be used as an identifier
+                # TODO: Noise duplicate_with_guardian normally when record IDs
+                # are implemented (MIC-4039)
+                data = data.map_partitions(unnoise_data)
+
+        # remove prenoised and missingness columns from data
```

Comment:
nit: Add a comment providing context, i.e. "now that we have the fully noised dataset, let's ..."